### PR TITLE
feat: Snyk Learn support via command [HEAD-114]

### PR DESCRIPTION
**Pull request by bastiandoetsch**
### Description

- This adds Snyk Learn support via the `snyk.openLearnLesson` and `snyk.getLearnLesson` commands. 
- A configuration is prepared for enabling code actions for opening the corresponding learn lesson. If it is set to true, the code actions are added to open source and code issues.
- The Snyk Learn API is queried for lessons and the lessons afterwards filtered by Ecosystem, Rule, CWE and CVE as criteria. If a single lessons can be determined, it is returned.
- Zerolog now uses the [LSP](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#window_logMessage) `window/logMessage` method to log, in addition to potential file logging. If sending the message to the server fails, zerolog uses stderr instead.

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [x] README.md updated, if user-facing
- [x] License file updated, if new 3rd-party dependency is introduced

### Screenshots / GIFs

image

https://user-images.githubusercontent.com/20150761/233145606-a1d4ca3f-8128-492a-95d0-b8aa58141093.mov

Log messages in vscode:
image

Log messages in Eclipse:
image

### CODE REVIEW

File: application/server/execute_command.go [hunk 1]
```diff
@@ -37,22 +39,24 @@ func executeCommandHandler(srv *jrpc2.Server) jrpc2.Handler {
 		// so we don't want to propagate it to functions that start background operations
 		bgCtx := context.Background()
 		method := "ExecuteCommandHandler"
+		c := config.CurrentConfig()
+		learnService := learn.New(c, c.Engine().GetNetworkAccess().GetUnauthorizedHttpClient)
```
Comment (PeterSchafer) [inline application/server/execute_command.go:43]: question: is it correct to use GetUnauthorizedHttpClient() why shouldn't we use a client with authorization?

File: domain/ide/command/get_learn_lesson.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	learnService learn.Service
+}
+
+func (cmd *getLearnLesson) Command() snyk.CommandData {
+	return cmd.command
+}
+
+func (cmd *getLearnLesson) Execute(_ context.Context) (any, error) {
+	args := cmd.command.Arguments
+	if len(args) < 4 {
```
Comment (PeterSchafer) [inline domain/ide/command/get_learn_lesson.go:43]: question: 5?

File: application/server/execute_command.go (hunk 1 above)
Comment (bastiandoetsch) [reply application/server/execute_command.go:43]: Learn doesn't need authorization :)

File: infrastructure/learn/service.go [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+						filteredLessons = append(filteredLessons, lesson)
+					}
+				}
+			}
+		}
+	}
+	return filteredLessons
+}
+
+func (s *serviceImpl) filterForCVEs(lessons []Lesson, cves []string) (filteredLessons []Lesson) {
```
Comment (PeterSchafer) [inline infrastructure/learn/service.go:291]: suggestion: combine filterForCWEs() and ...CVEs()

**Review by PeterSchafer — APPROVED** [review]

**Pull request merged**